src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java

```diff
@@ -382,9 +380,48 @@ public void validateJudgementsHasJurisdiction(CaseData caseData, boolean partOfM
         return errors;
     }
 
+    public void validateHearingStatusForCaseCloseEvent(CaseData caseData, List<String> errors) {
+        for (var currentHearingTypeItem : caseData.getHearingCollection()) {
+            for (var currentDateListedTypeItem : currentHearingTypeItem.getValue().getHearingDateCollection()) {
+                if (HEARING_STATUS_LISTED.equals(currentDateListedTypeItem.getValue().getHearingStatus())) {
+                    errors.add(CLOSING_LISTED_CASE_ERROR);
+                    break;
+                }
+            }
+        }
+    }
+
+    public void validateHearingJudgeAllocationForCaseCloseEvent(CaseData caseData, List<String> errors) {
+        for (var currentHearingTypeItem : caseData.getHearingCollection()) {
+            if (currentHearingTypeItem.getValue().getJudge() == null) {
```

Comment:
ECM-407: Not checking if hearingStatus = Heard.

Reply:
should check to see if collection exists? would we not have a null pointer here if they tried to close the case without having any hearings?

Reply:
Thanks. Having a look.

Reply:
fixed